### ISSUE TITLE
lxc-checkconfig wrong report on memory controller for linux 4.x [Fix Included]

### ISSUE THREAD

**Issue by diego-XA** [lxc/lxc#508]
The check for "memory controller" in lxc-checkconfig is buggy. Currently it checks this way:
```bash
if [ $KVER_MAJOR -ge 3 -a $KVER_MINOR -ge 6 ]; then
    is_enabled CONFIG_MEMCG
else
    is_enabled CONFIG_CGROUP_MEM_RES_CTLR
fi
```

Maybe this could work better:
```bash
if [ $KVER_MAJOR -le 3 -a $KVER_MINOR -lt 6 ]; then
    is_enabled CONFIG_CGROUP_MEM_RES_CTLR
else
    is_enabled CONFIG_MEMCG
fi
```
With this last code, testing in my Linux 4.0  server, the memory controller is reported as enabled. Then, I forced the KVER_MAJOR and KVER_MINOR for different versions to test it:
```bash
KVER_MAJOR=3
KVER_MINOR=9
====> Cgroup memory controller: enabled 

KVER_MAJOR=3
KVER_MINOR=6
====> Cgroup memory controller: enabled 

KVER_MAJOR=3
KVER_MINOR=5
====> Cgroup memory controller: missing
```
So I believe the code works.

**Comment by brauner** [comment]
Fix present as pull request.

**Comment by diego-XA** [comment]
I know it is only tiny bit of code, but I dont understand why you did not used my code and implemented a new one which is far more complex and also less efficient... I created my own pull request.

**Comment by brauner** [comment]
Simply because my  pull request was made a few days before yours. It's difficult to jump timelines.

**Comment by diego-XA** [comment]
Well, you are right, I didn't realise that your pull request was a week ago. I just didn't understand.
In such case, please feel free to use the above code if you think it might be a better option. In my opini it is, but it may be a matter of tastes :).

**Comment by brauner** [comment]
No problem. Life without misunderstandings would be boring. :)

**Comment by davidfavor** [comment]
I hit this same bug a few days ago.

https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1471358 contains a patch file with one way to fix this.

**Comment by diego-XA** [comment]
This bug should be marked as fixed as it was already patched upstream.

**Comment by Kubuxu** [comment]
It was not fixed.

**Comment by alkino** [comment]
It was fixed by #498. But this PR is not in 1.1.2, you should wait the next release if you want it upstream.
